test/controllers/products_controller_test.rb

```diff
@@ -0,0 +1,22 @@
+require 'test_helper'
+
```

Comment:
I'm used to RSpec so I can't give syntax level feedback.

This is a good start!

You're going to want to have tests for each public method in the controller.
And whenever there is an if/else you will want to consider each path to make sure both sides are covered.